bindings/python/fdb/directory_impl.py

```diff
@@ -613,17 +612,17 @@ def _is_prefix_empty(self, tr, prefix):
 
 def _to_unicode_path(path):
     if isinstance(path, bytes):
-        path = six.text_type(path)
+        path = str(path)
```

Comment:
Hmm, happy to change it, but wouldn't they be the same? Both `str(path)` or `path.decode()` would return a unicode string in Python 3, no? 🤔

Reply:
Oh I see, you are absolutely right. Continuing on your answer,

```python
>>> s.decode() == str(s)
False
```

But for `str` to decode `bytes`, `encoding` must be passed:

```python
>>> s.decode() == str(s, 'utf-8')
True
```

Also validated with the docs:

> If neither encoding nor errors is given, str(object) returns [type(object).__str__(object)](dfile:///Users/sepeth/Library/Application%20Support/Dash/DocSets/Python_3/Python%203.docset/Contents/Resources/Documents/doc/reference/datamodel.html#object.__str__), which is the “informal” or nicely printable string representation of object. For string objects, this is the string itself. If object does not have a [__str__()](dfile:///Users/sepeth/Library/Application%20Support/Dash/DocSets/Python_3/Python%203.docset/Contents/Resources/Documents/doc/reference/datamodel.html#object.__str__) method, then [str()](dfile:///Users/sepeth/Library/Application%20Support/Dash/DocSets/Python_3/Python%203.docset/Contents/Resources/Documents/doc/library/stdtypes.html#str) falls back to returning [repr(object)](dfile:///Users/sepeth/Library/Application%20Support/Dash/DocSets/Python_3/Python%203.docset/Contents/Resources/Documents/doc/library/functions.html#repr).

https://docs.python.org/3/library/stdtypes.html#str

TIL, and I will make the change.

May you please also have a look at the comment for ` _to_unicode_path`?

Reply:
Yeah, Python has plenty of pitfalls but better than C++, while I would suggest we move to Rust sometime since things would be more explicit -- in Rust str must be encoded and bytes are simply [u8]; so you can never simply [u8].to_str() without checking the encoding.

Reply:
Yeah ^-^ Also TIL, `python -b` to catch this. Apparently, it is a thing, and there's also `python -bb` to treat these as errors.